src/bench_ecmult.c

```diff
@@ -34,15 +35,144 @@ typedef struct {
     size_t count;
     int includes_g;
 
-    /* Changes per test iteration */
+    /* Changes per test iteration, used to pick different scalars and pubkeys
+     * in each run. */
     size_t offset1;
     size_t offset2;
 
     /* Test output. */
     secp256k1_gej* output;
 } bench_data;
 
-static int bench_callback(secp256k1_scalar* sc, secp256k1_ge* ge, size_t idx, void* arg) {
+/* Hashes x into [0, POINTS) twice and store the result in offset1 and offset2. */
+static void hash_into_offset(bench_data* data, size_t x) {
+    data->offset1 = (x * 0x537b7f6f + 0x8f66a481) % POINTS;
+    data->offset2 = (x * 0x7f6f537b + 0x6a1a8f49) % POINTS;
```

Comment:
where's this thing from?
I see it's also in `bench_ecmult_setup` but can't find the source of this

Reply:
I think is this is just a standard integer hash function (akin to https://en.wikipedia.org/wiki/Universal_hashing#Hashing_integers) whose parameters sipa selected. But looking at it now it may be give poor results because `POINTS` is not prime.

Reply:
I don't think points needs to be prime, so long as the multipliers are coprime to it.

Reply:
fwiw they're all coprime to POINTS.